_rules/SC2-4-6-descriptive-headings.md

```diff
@@ -0,0 +1,134 @@
+---
+name: Descriptive headings
+test_type: atomic
+
+description: |
+   This rule checks that headings describe the topic or purpose of the content.
+   
+success_criterion:
+- 2.4.6 # Headings and labels
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Dagfinn Rømen
+- Geir Sindre Fossøy
+- Carlos Duarte
+---
+
+## Test procedure
+
+### Applicability
+
+The rule applies to any element with the the [semantic role](#semantic-role) of `heading`.
```

Comment:
Suggest adding either: 

- "that is [visible](#visible) and [included in the accessibility tree](#included-in-the-accessibility-tree)" (AND), or 
- "that is either [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree)" (OR)

Reply:
I would suggest to add the second : visible OR included in the accessibility tree. If I understand all the definitions, this include cases where headings are off screen (so not visible) but still in the accessibility tree. Is that it @annethyme ?

Reply:
Hum, but thinking a little more...
For cases where heading is visible but has an `aria-hidden="true"` (so not in the accessibility tree anymore), the rule should not apply. 
So, in the end, I suggest that's the heading just has to be included in the accessibility tree, the visibility condition is not necessary ? 

> "that is included in the [included in the accessibility tree](#included-in-the-accessibility-tree)"

Reply:
Good point, it makes sense to test only headings that the user can access. We have added "included in the accessibility tree".